### PR TITLE
Add support for a `hybrid` object

### PR DESCRIPTION
**Pull request by fyockm**
Provide all methods for both object and subject on a single schema

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,12 +94,25 @@ UserSchema.plugin(acl.subject, {
 });
 ```
 
+### Hybrid
+
+Combines Subject and Object, so that a subject can introspectively see if has permissions on itself. `getAccess` and `setAccess` method on the subject are renamed as `getSubjectAccess` and `setSubjectAccess`, respectively. All other option/method names remain the same.
```
Comment (scttnlsn) [inline README.md]: Typo here: `see [if] it has`

Maybe also change to "combines `subject` and `object`".

Could you include an example of registering the `hybrid` plugin on a schema?

File: package.json [hunk 2]
```diff
@@ -1,27 +1,25 @@
...
-    "devDependencies": {
-        "mocha": ">= 1.4.1",
-        "mongoose": ">= 3.1.0",
-        "sinon": ">= 1.4.2"
-    }
-}
\ No newline at end of file
+  "name": "mongoose-acl",
+  "description": "Mongoose ACL",
+  "version": "1.0.0",
```
Comment (scttnlsn) [inline package.json]: Could you revert this change to the version number?  I'll bump before publishing to NPM.

**Comment by scttnlsn** [comment]
Thanks @fyockm.

I left a couple comments inline.  I also think duplicating all those tests is a little unnecessary.  Instead, could we just register `hybrid` on an object and assert somehow that the results of calling `getSubjectAccess` and `setSubjectAccess` are the same as when calling `getAccess` and `setAccess` explicitly on the subject?

I'm also not thoroughly convinced that `hybrid` is the best name for this.

**Comment by fyockm** [comment]
@scttnlsn thanks for the feedback. I will update the README and revert the version number.

For the tests, I was just trying to show that a hybrid would pass all the subject *and* object tests. But perhaps there is a better way, as you suggest. I'll take another crack at it.

I'm certainly open to naming recommendations. `hybrid` was just the first thing that popped in my head. Here are a couple other ideas:
- sobject _(kidding)_
- entity
- compound
- resource
Do any of those speak to you?

**Comment by scttnlsn** [comment]
Great, thanks!

I don't really have any better name suggestions.  It feels more like something self-referential or circular though rather than the hybrid of two concepts.  Don't worry about the name.

**Comment by fyockm** [comment]
@scttnlsn please let me know if I missed anything.

**Comment by scttnlsn** [comment]
Thanks!

**Comment by scttnlsn** [comment]
@fyockm I think I'm going to update the example in the README:

```javascript
UserSchema.plugin(acl.hybrid);

user.setAccess('*', ['read']);
user.setSubjectAccess(user, ['write', 'delete']);
```

Is this the sort of use case you had in mind?

**Comment by fyockm** [comment]
@scttnlsn yep, that's exactly how I am using.

**Comment by scttnlsn** [comment]
Okay, thanks.

This is released in v0.2.2

**Pull request merged**